drivers/gpio/gpio_ifx_cat1.c

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (c) 2022 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief GPIO driver for Infineon CAT1 MCU family.
+ *
+ * Note:
+ * - Trigger detection on pin rising or falling edge (GPIO_INT_TRIG_BOTH)
+ *   is not supported in current version of GPIO CAT1 driver.
+ */
+
+#define DT_DRV_COMPAT  infineon_cat1_gpio
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#include <cyhal_gpio.h>
+#include <cyhal_hwmgr.h>
+#include <cy_gpio.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(gpio_cat1, CONFIG_GPIO_LOG_LEVEL);
+
+/* Device config structure */
+struct gpio_cat1_config {
+	/* gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+	cyhal_gpio_callback_data_t *cb_data_ptr;
+	GPIO_PRT_Type              *regs;
+	uint8_t ngpios;
+	uint8_t intr_priority;
+};
+
+/* Data structure */
+struct gpio_cat1_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data common;
+
+	/* device's owner of this data */
+	const struct device *dev;
+
+	/* callbacks list */
+	sys_slist_t callbacks;
+};
+
+/* Get port number by calculation difference from current port address minus
+ * GPIO base address divided by GPIO structure size.
+ */
+#define GET_PORT_NUM(dev)				     \
+	(((uint32_t) ((const struct gpio_cat1_config *const) \
+		      (dev)->config)->regs - CY_GPIO_BASE) / GPIO_PRT_SECTION_SIZE)
+
+#define GET_DEV_OBJ_FROM_LIST(i, _) \
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio_prt##i))
+
+/* Map port number to device object */
+static const struct device *const port_dev_obj[IOSS_GPIO_GPIO_PORT_NR] = {
+	LISTIFY(15, GET_DEV_OBJ_FROM_LIST, ( , ))
```

Comment:
```suggestion
	LISTIFY(15, GET_DEV_OBJ_FROM_LIST, (,))
```
this should suppress following warning (and it will make consistent with other code in Zephyr):
```
SPACING
drivers/gpio/gpio_ifx_cat1.c:62  space prohibited after that open parenthesis '('
```

Reply:
updated